### PR TITLE
Allowing admin users to change users email

### LINKED ISSUE #2647: Admin can't change users email address
### What happened?

Editing a user account from the admin control panel doesn't allow the admin to change their email. The field is grayed out and not editable.

image

### What did you expect?

To change the email on file.

### Version

0.7.13.2

### What OS is Kavita being run on?

Docker

### If the issue is being seen on Desktop, what OS are you running where you see the issue?

None

### If the issue is being seen in the UI, what browsers are you seeing the problem on?

_No response_

### If the issue is being seen on Mobile, what OS are you running where you see the issue?

None

### If the issue is being seen on the UI, what browsers are you seeing the problem on?

_No response_

### Relevant log output

_No response_

### Additional Notes

.

### PR DESCRIPTION
**Pull request by DieselTech**
# Changed
- Changed: Allowing admins to edit user profiles to update their email addresses. (Closes https://github.com/Kareadita/Kavita/issues/2647)

### CODE REVIEW

File: API/Controllers/AccountController.cs [hunk 1]
```diff
@@ -508,6 +508,19 @@ public async Task<ActionResult> UpdateAccount(UpdateUserDto dto)
...
             _unitOfWork.UserRepository.Update(user);
         }
+    
+        // Update email
+        if (!string.IsNullOrEmpty(dto.Email) && !user.Email!.Equals(dto.Email))
+        {
+            // Validate email change
+            var emailValidationErrors = await _accountService.ValidateEmail(dto.Email);
+            if (emailValidationErrors.Any()) return BadRequest(await _localizationService.Translate(User.GetUserId(), "email-taken"));
+            user.Email = dto.Email;
```
Comment (majora2007) [inline API/Controllers/AccountController.cs:519]: You need a new line after the if statement to help the code be easier to digest.

File: API/Controllers/AccountController.cs [hunk 2]
```diff
@@ -508,6 +508,19 @@ public async Task<ActionResult> UpdateAccount(UpdateUserDto dto)
...
+            user.EmailConfirmed = false;
+            user.NormalizedEmail = _userManager.NormalizeEmail(dto.Email);
+            user.ConfirmationToken = await _userManager.GenerateEmailConfirmationTokenAsync(user);
```
Comment (majora2007) [inline API/Controllers/AccountController.cs]: You generate a confirmation token, but I don't see you triggering an email for the end user to confirm and get rid of that pending status.

**Review by majora2007 — CHANGES_REQUESTED** [review]
Just did a preliminary review. I don't see any of the email code to trigger a link so the user can confirm the email change.

**Comment by majora2007** [comment]
Not sure if you still want to work on this to get it in?

**Comment by DieselTech** [comment]
Changes made. Got rid of the email confirmation step because it is an admin making the change. That should force change it and not rely on the confirmation flow. Not sure what I was thinking before.

**Comment by majora2007** [comment]
I'm going to implement this myself, it's going to be much easier. I'll dm you the correct way to implement this.

**Comment by majora2007** [comment]
Here is the correct way to implement:
https://github.com/Kareadita/Kavita/pull/3446/commits/d522bccf86cb11efd1b59e380bbbc446c2c6ba35

**Pull request closed without merging**